collect_app/src/main/java/org/odk/collect/android/tasks/DownloadFormsTask.java

```diff
@@ -68,6 +68,10 @@
 public class DownloadFormsTask extends
         AsyncTask<ArrayList<FormDetails>, String, HashMap<FormDetails, String>> {
 
+    class CancelDownloads extends Exception {
```

Comment:
I'm generally against including "type of thing" info in names but it seems Java is pretty serious about adding `Exception` to the end of all its exception names so this feels weird!

Is the exception really needed? Seems the method could just return early instead of throwing?

Reply:
In Scala I would use an Option or an Either to allow me to return one of three states:
- it worked
- it didn’t work and here’s why
- this one didn’t finish and don’t do any more

In Java (7) I could have done this with 
- empty string
- string with errors
- null

But it didn’t seem right.

It’s interesting to note that IDEA’s Extract Method refactoring replaced each `break` with `return null`, and modified the caller to break if null was returned. Clever!

If we keep the exception, I’m happy to add `Exception` to the end of the name. Time was, I would have insisted on it myself, but my time with Scala really changed me.

Reply:
Ah, of course. For some reason I was thinking the `throw`s were in void methods. Fine with using an exception, then. There's already `TaskCancelledException` in the `exception` package. Would be great to use that if possible.

Reply:
Good idea about using TaskCancelledException.